get_started.md

```diff
@@ -6,66 +6,181 @@ SPDX-License-Identifier: Apache-2.0
+The credentials secret will be used by the device to authenticate with Astarte.
+Copy it somewhere safe as it will be used in the next steps.
+
+## Installin the Astarte device SDK for Python
+
+The library source code can be found on
+[GitHub](https://github.com/astarte-platform/astarte-device-sdk-python) and can be installed using
+pip:
+```sh
+pip install astarte-device-sdk
 ```
 
-## Registering a device
+## Creating a device instance
 
-In order for a device to connect to Astarte a registration procedure is required. This registration
-will produce a device specific credential secret that will be used when connecting to Astarte.
-The Python Astarte device SDKs provides utilities to perform a device registration directly on
-the device. Those utilities require a registration JWT to be loaded on the device. Such JWT should
-be discarded following the registration procedure.
+The first step to use the Astarte device SDK is to import the device class from the astarte package
+and create a device instance.
 
-This step is only useful when registering the device through the APIs using the JWT token.
-Registration of a device can also be performed outside the device in the Astarte instance using
-tools such as [astartectl](https://github.com/astarte-platform/astartectl), the
-[Astarte dashboard](https://docs.astarte-platform.org/astarte/latest/015-astarte_dashboard.html),
-or the dedicated
-[Astarte API for device registration](https://docs.astarte-platform.org/astarte/latest/api/index.html?urls.primaryName=Pairing%20API).
-The generated credential secret should then be loaded manually on the device.
+The init function for the device class will require some information:
+- The Astarte pairing base URL. This is the Astarte base URL with an appendend `/pairing` at the
+  end. If you deployed an Astarte instance using the Astarte quick instance guide and decided to
+  use the SDK on the same machine as the Astarte instance, this will be
+  `http://api.astarte.localhost/pairing`.
+- The name of the realm where the device is registered. If you deployed an Astarte instance using
+  the Astarte quick instance guide, you can use the default generated one named `test`.
+- The device ID.
+- The credentials sected associated to the device ID.
+- The path to a directory where the Astarte device will store some persistent data.
 
-To generate a credential secret use the function `register_device_with_jwt_token` as shown below.
 ```python
-from astarte.device import register_device_with_jwt_token
-
-DEVICE_ID = "<DEVICE_ID>"
-REALM = "<REALM>"
-JWT_TOKEN = "<JWT_TOKEN>"
-PAIRING_BASE_URL = "<PAIRING_BASE_URL>"
+import os
+from astarte.device import DeviceMqtt
 
-credentials_secret = register_device_with_jwt_token(DEVICE_ID, REALM, JWT_TOKEN, PAIRING_BASE_URL)
+astarte_pairing_base_url = "http://api.astarte.localhost/pairing"
+realm = "test"
+device_id = "MFVgjjP2Tt6RbT_wKOI0VA"
```

Comment:
Use generic values
```suggestion
astarte_pairing_base_url = "<PAIRING_BASE_URL>"
realm = "<REALM>"
device_id = "<DEVICE_ID>"
```

Reply:
I left the defaults in there so the user can make an easy copy-paste of the snippet.

Reply:
even for the `device_id`?

Reply:
Thinking back I agree with you, it makes more sense to keep them generic